core/src/main/java/org/elasticsearch/discovery/local/LocalDiscovery.java

```diff
@@ -305,6 +285,11 @@ public DiscoveryStats stats() {
     }
 
     @Override
+    public DiscoverySettings getDiscoverySettings() {
+        return discoverySettings;
+    }
+
```

Comment:
How about converting the `Discovery` interface into an `AbstractDiscovery` class which subsumes `DiscoverySettings`?